src/realm.js

```diff
@@ -271,6 +282,45 @@ export class Realm {
     }
   }
 
+  add_prior_effects(priorEffects: Effects, subsequentEffects: Effects) {
+    let [pc, pg, pb, pp, po] = priorEffects;
+    let [sc, sg, sb, sp, so] = subsequentEffects;
+
+    pc; sc;
+
+    let saved_generator = this.generator;
+    this.generator = pg.clone();
```

Comment:
Oops yes, I do need to build a new generator since it gets modified and may get modified more than once, but it certainly makes zero sense to throw it away.